sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -244,9 +235,28 @@ public Response<T> WaitForCompletion(CancellationToken cancellationToken)
         /// <returns>The last HTTP response received from the server, including the final result of the long-running operation.</returns>
         /// <exception cref="RequestFailedException">Thrown if there's been any issues during the connection, or if the operation has completed with failures.</exception>
         public Response<T> WaitForCompletion(TimeSpan pollingInterval, CancellationToken cancellationToken)
+            => WaitForCompletionAsync(async: false, pollingInterval, cancellationToken).EnsureCompleted();
+
+        private async ValueTask<Response<T>> WaitForCompletionAsync(bool async, TimeSpan? pollingInterval, CancellationToken cancellationToken)
```

Comment:
It looks like that if you add `scopeName` parameter to `OperationInternalBase.WaitForCompletionResponseAsync`, this method can be reduced to:
```cs
private async ValueTask<Response<T>> WaitForCompletionAsync(bool async, TimeSpan? pollingInterval, CancellationToken cancellationToken) {
    var rawResponse = await WaitForCompletionResponseAsync(async, pollingInterval, _waitForCompletionScopeName, cancellationToken);
    return Response.FromValue(Value, rawResponse);
}
```

Reply:
`WaitForCompletionResponseAsync` can only have scope name `_waitForCompletionResponseScopeName`. I don't think it is a good practice to expose this parameter.